tests/workbenches/utils.py

```diff
@@ -1,11 +1,23 @@
-from kubernetes.dynamic import DynamicClient, Resource, ResourceInstance
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.self_subject_review import SelfSubjectReview
+from ocp_resources.user import User
+from simple_logger.logger import get_logger
 
+LOGGER = get_logger(name="workbenches_utils")
 
-def get_username(dyn_client: DynamicClient) -> str:
+
+def get_username(dyn_client: DynamicClient) -> str | None:
     """Gets the username for the client (see kubectl -v8 auth whoami)"""
-    self_subject_review_resource: Resource = dyn_client.resources.get(
-        api_version="authentication.k8s.io/v1", kind="SelfSubjectReview"
-    )
-    self_subject_review: ResourceInstance = dyn_client.create(self_subject_review_resource)
-    username: str = self_subject_review.status.userInfo.username
+    username: str | None
+    try:
+        self_subject_review = SelfSubjectReview(client=dyn_client, name="selfSubjectReview").create()
+        assert self_subject_review
+        username = self_subject_review.status.userInfo.username
+    except NotImplementedError:
+        LOGGER.info(
+            "SelfSubjectReview not found. Falling back to user.openshift.io/v1/users/~ for OpenShift versions <=4.14"
+        )
+        user = User(client=dyn_client, name="~").instance
+        username = user.get("metadata", {}).get("name", None)
```

Comment:
Should you check if user exists before using ".instance"?

Reply:
On Openshift, `~` user always exists

Reply:
This seems to be equivalent of `oc whoami`

Reply:
Thanks for the review.

If I wanted to use both the admin_client and unprivileged_client in the same test, I don't know how I'd be switching between them so that the `oc whoami` runs with the correct context.

So I thought the current way is actually simpler.

https://github.com/opendatahub-io/opendatahub-tests/blob/eb7987221ef543a399baa0502cb710f9c029582c/tests/conftest.py#L307